CMakeLists.txt

```diff
@@ -134,7 +134,7 @@ ENDIF()
 
 IF(WITH_VIGRANUMPY)
     set(WITH_BOOST "ON")
-    # set(WITH_BOOST_COMPONENTS ${WITH_BOOST_COMPONENTS} "python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+    set(WITH_BOOST_COMPONENTS ${WITH_BOOST_COMPONENTS} "python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
```

Comment:
I don't follow how that relates to adding Boost.Python. In the current strategy, we are not looking for Boost.Python component and not guaranteeing it was build against the right version of Python. I have found that not setting this will cause us to pick up the wrong Boost.Python instead.

By making this change, I was able to discover that we were picking up the wrong Boost.Python (namely 3.2 when trying to test 3.4 or 3.5). Once added it would simply skip testing on Python with versions 3.4 or 3.5 because it couldn't find a compatible copy of Boost.Python. All of this was possible because this constraint ensures only a copy Boost.Python that works with our interpreter is picked.